File: algorithm/leetcode1283.find-the-smallest-divisor-given-a-threshold.cpp
class Solution {
public:
    int smallestDivisor(vector<int>& nums, int threshold) {
    	int i, l = 1, r = 1000000, mid, sum;
    	while(l < r)
    	{
    		mid = l+((r-l)>>1);
    		sum = 0;
    		for(i = 0; i < nums.size(); ++i)
	    		sum += ceil(nums[i]/double(mid));
    		if(sum > threshold)
    			l = mid+1;
    		else
    			r = mid;
    	}
    	return l;
    }
};

File: algorithm/Python3_version/leetcode1283.find-the-smallest-divisor-given-a-threshold.py
class Solution:
    def smallestDivisor(self, nums: List[int], threshold: int) -> int:
        l, r = 1, 1000000
        n = len(nums)
        while l < r:
            mid = l+((r-l)>>1)
            sum = 0
            for i in range(n):
                sum += ceil(nums[i]/mid)
            if sum > threshold:
                l = mid+1
            else:
                r = mid
        return l